front/components/assistant/AssistantPicker.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Button,
+  DropdownMenu,
+  MagnifyingGlassStrokeIcon,
+  PlusIcon,
+  RobotIcon,
+  WrenchIcon,
+} from "@dust-tt/sparkle";
+import Link from "next/link";
+import { useState } from "react";
+
+import { AgentConfigurationType } from "@app/types/assistant/agent";
+import { WorkspaceType } from "@app/types/user";
+
+export function AssistantPicker({
+  owner,
+  assistants,
+  onItemClick,
+  pickerButton,
+  manageButtons,
+}: {
+  owner: WorkspaceType;
+  assistants: AgentConfigurationType[];
+  onItemClick: (assistant: AgentConfigurationType) => void;
+  pickerButton?: React.ReactNode;
+  manageButtons?: boolean;
+}) {
+  const [searchText, setSearchText] = useState("");
+  const searchedAssistants = assistants.filter((a) =>
+    a.name.toLowerCase().startsWith(searchText.toLowerCase())
+  );
+  return (
+    <DropdownMenu>
+      <div onClick={() => setSearchText("")}>
+        {pickerButton ? (
+          <DropdownMenu.Button>{pickerButton}</DropdownMenu.Button>
+        ) : (
+          <DropdownMenu.Button icon={RobotIcon} />
+        )}
+      </div>
+      <DropdownMenu.Items origin="auto" width={240}>
+        {assistants.length > 7 && (
+          <div className="border-b border-structure-100 p-2">
+            <div className="relative text-sm font-medium text-element-800">
+              <input
+                type="text"
+                placeholder="Search"
+                value={searchText}
+                className="h-9 w-full rounded-full border-structure-200 pr-8"
+                onKeyDown={(e) => {
+                  if (e.key === "Enter" && searchedAssistants.length > 0) {
+                    onItemClick(searchedAssistants[0]);
+                    setSearchText("");
+                  }
+                }}
+                onChange={(e) => {
+                  setSearchText(e.target.value);
+                }}
+                autoFocus={true}
+              />
+              <MagnifyingGlassStrokeIcon className="absolute right-3 top-2 h-5 w-5" />
+            </div>
+          </div>
+        )}
+        <div className="max-h-[22.5rem] overflow-y-auto [&>*]:w-full">
```

Comment:
what is `[&>*]:w-full` ?

Reply:
indicates that Dropdown Items should be full width. We have to put them in a div for the max-width / scroll that should only apply to items (not search above nor buttons below) but then the styling made by Item*s* does not propagate

It's not completely satisfying I agree but better than alternatives I could think of without completely rethinking the dropdown component